MemoryManagement/src/memorymanagementAlgorithm/Memory.java

```diff
@@ -55,37 +68,40 @@ public int get_total_free() {
     //====================== method sections ====================//
     public void sort_segment_vector(Vector<Segment> input) {
         Collections.sort(input, (a, b) -> a.getBase() < b.getBase() ? -1 : a.getBase() == b.getBase() ? 0 : 1);
+
     }
 
-    public void swap_base_2_segments(Segment first, Segment second) {
+    private void swap_base_2_segments(Segment first, Segment second) {

```

Comment:
This is a wrong function implementation or wrong named.
suppose 2 segments that don't lay after each other.

Reply:
it's for sorting just to swap the main function handle the logic .
not fixed